splunk_eventgen/eventgen_api_server/eventgen_core_object.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import logging
+import os
+
+import splunk_eventgen.eventgen_core as eventgen_core
+
+FILE_PATH = os.path.dirname(os.path.realpath(__file__))
+CUSTOM_CONFIG_PATH = os.path.realpath(os.path.join(FILE_PATH, "..", "default", "eventgen_wsgi.conf"))
+
+class EventgenCoreObject():
+    def __init__(self):
+        self.logger = logging.getLogger('eventgen_server')
+        self.eventgen_core_object = eventgen_core.EventGenerator(self._create_args())
+        self.configured = False
+        self.configfile = None
+        self.check_and_configure_eventgen()
+    
+    def check_and_configure_eventgen(self):
+        if os.path.isfile(CUSTOM_CONFIG_PATH):
+            self.configured = True
+            self.configfile = CUSTOM_CONFIG_PATH
+            self.eventgen_core_object.reload_conf(CUSTOM_CONFIG_PATH)
+            self.logger.info("Configured Eventgen from {}".format(CUSTOM_CONFIG_PATH))
+
+    def refresh_eventgen_core_object(self):
+        self.eventgen_core_object = eventgen_core.EventGenerator(self._create_args())
+        self.configured = False
+        self.configfile = None
+        self.check_and_configure_eventgen()
+        self.logger.info("Refreshed the eventgen core object")
+
+    def _create_args(self):
+        args = argparse.Namespace()
+        args.daemon = False
+        args.verbosity = None
+        args.version = False
+        args.backfill = None
+        args.count = None
+        args.devnull = False
+        args.disableOutputQueue = False
+        args.end = None
+        args.generators = None
+        args.interval = None
+        args.keepoutput = False
+        args.modinput = False
+        args.multiprocess = True
```

Comment:
it will use `multiprocess` by default?

Reply:
I am on a debate about this. I would rather have Eventgen server to work in one mode to keep the implementation consistent. It is just too much hassle to support both threading and multiprocessing. In this situation, I would support multiprocessing by default, because the use case is more wide.